[CI Failure]: [Kernels (B200)

### Name of failing test

`tests/models/quantization/test_nvfp4.py::test_nvfp4[flashinfer_trtllm-True-nvidia/Llama-3.1-8B-Instruct-NVFP4] 

### Basic information

- [ ] Flaky test
- [ ] Can reproduce locally
- [ ] Caused by external libraries (e.g. bug in `transformers`)

### 🧪 Describe the failing test

[url](https://buildkite.com/vllm/ci/builds/66835/list?sid=019e3ed4-2546-4324-88d2-38d6305b8f60&tab=output)

```
=========================================================================== FAILURES ===========================================================================
____________________________________________ test_nvfp4[flashinfer_trtllm-True-nvidia/Llama-3.1-8B-Instruct-NVFP4] _____________________________________________
vllm_runner = <class 'tests.conftest.VllmRunner'>, model = 'nvidia/Llama-3.1-8B-Instruct-NVFP4', eager = True, backend = 'flashinfer_trtllm'
    @pytest.mark.parametrize("model", ["nvidia/Llama-3.1-8B-Instruct-NVFP4"])
    @pytest.mark.parametrize("eager", EAGER)
    @pytest.mark.parametrize(
        "backend",
        [
            "emulation",
            "flashinfer_cudnn",
            "flashinfer_trtllm",  # the small seq_len ensures trtllm_8x4_layout backend is used
            "flashinfer_cutlass",
        ],
    )
    def test_nvfp4(vllm_runner, model, eager, backend):
        if (
            not current_platform.has_device_capability(100)
            and backend in SM_100_NVFP4_BACKENDS
        ):
            pytest.skip(
                f"The backend {backend} is not supported with current_platform.has_device_capability(100) == False"
            )
>       with vllm_runner(model, enforce_eager=eager, linear_backend=backend) as llm:
tests/models/quantization/test_nvfp4.py:119:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/conftest.py:923: in __init__
    self.llm = LLM(
/usr/local/lib/python3.12/dist-packages/vllm/entrypoints/llm.py:375: in __init__
    self.llm_engine = LLMEngine.from_engine_args(
/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/llm_engine.py:170: in from_engine_args
    return cls(
/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/llm_engine.py:104: in __init__
    self.engine_core = EngineCoreClient.make_client(
/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core_client.py:101: in make_client
    return SyncMPClient(vllm_config, executor_class, log_stats)
/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core_client.py:723: in __init__
    super().__init__(
/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core_client.py:535: in __init__
    with launch_core_engines(
/usr/lib/python3.12/contextlib.py:144: in __exit__
    next(self.gen)
/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/utils.py:1133: in launch_core_engines
    wait_for_engine_startup(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
handshake_socket = <zmq.Socket(zmq.ROUTER) at 0x7286bd7f64a0 closed>
addresses = EngineZmqAddresses(inputs=['ipc:///tmp/2339eccd-e5db-4320-a23d-4133acef7791'], outputs=['ipc:///tmp/bd46a73b-e999-4f1c-9be8-cc19a573a463'], coordinator_input=None, coordinator_output=None, frontend_stats_publish_address=None)
core_engines = [<vllm.v1.engine.utils.CoreEngine object at 0x7287275a3fe0>]
parallel_config = ParallelConfig(pipeline_parallel_size=1, tensor_parallel_size=1, prefill_context_parallel_size=1, data_parallel_size=1..._comm_backend='ag_rs', cp_kv_cache_interleave_size=1, data_parallel_index=0, _api_process_count=1, _api_process_rank=0)
coordinated_dp = False
cache_config = CacheConfig(block_size=16, user_specified_block_size=True, user_specified_mamba_block_size=False, hash_block_size=None...=False, kv_cache_memory_bytes=None, kv_offloading_size=None, kv_offloading_backend='native', _block_size_resolved=True)
proc_manager = <vllm.v1.engine.utils.CoreEngineProcManager object at 0x728717fa37a0>, coord_process = None
    def wait_for_engine_startup(
        handshake_socket: zmq.Socket,
        addresses: EngineZmqAddresses,
        core_engines: list[CoreEngine],
        parallel_config: ParallelConfig,
        coordinated_dp: bool,
        cache_config: CacheConfig,
        proc_manager: CoreEngineProcManager | None,
        coord_process: Process | None,
    ):
        # Wait for engine core process(es) to send ready messages.
        local_count = parallel_config.data_parallel_size_local
        remote_count = len(core_engines) - local_count
        # [local, remote] counts
        conn_pending, start_pending = [local_count, remote_count], [0, 0]
        poller = zmq.Poller()
        poller.register(handshake_socket, zmq.POLLIN)
        remote_should_be_headless = (
            not parallel_config.data_parallel_hybrid_lb
            and not parallel_config.data_parallel_external_lb
        )
        if proc_manager is not None:
            for sentinel in proc_manager.sentinels():
                poller.register(sentinel, zmq.POLLIN)
        if coord_process is not None:
            poller.register(coord_process.sentinel, zmq.POLLIN)
        while any(conn_pending) or any(start_pending):
            events = poller.poll(STARTUP_POLL_PERIOD_MS)
            if not events:
                if any(conn_pending):
                    logger.debug(
                        "Waiting for %d local, %d remote core engine proc(s) to connect.",
                        *conn_pending,
                    )
                if any(start_pending):
                    logger.debug(
                        "Waiting for %d local, %d remote core engine proc(s) to start.",
                        *start_pending,
                    )
                continue
            if len(events) > 1 or events[0][0] != handshake_socket:
                # One of the local core processes exited.
                finished = proc_manager.finished_procs() if proc_manager else {}
                if coord_process is not None and coord_process.exitcode is not None:
                    finished[coord_process.name] = coord_process.exitcode
>               raise RuntimeError(
                    "Engine core initialization failed. "
                    "See root cause above. "
                    f"Failed core proc(s): {finished}"
                )
E               RuntimeError: Engine core initialization failed. See root cause above. Failed core proc(s): {}
/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/utils.py:1192: RuntimeError
======================================================================= warnings summary =======================================================================
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute
../usr/local/lib/python3.12/dist-packages/torch/jit/_script.py:365: 14 warnings
  /usr/local/lib/python3.12/dist-packages/torch/jit/_script.py:365: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(
../usr/local/lib/python3.12/dist-packages/schemathesis/generation/coverage.py:305
  /usr/local/lib/python3.12/dist-packages/schemathesis/generation/coverage.py:305: DeprecationWarning: jsonschema.exceptions.RefResolutionError is deprecated as of version 4.18.0. If you wish to catch potential reference resolution errors, directly catch referencing.exceptions.Unresolvable.
    ref_error: type[Exception] = jsonschema.RefResolutionError,
tests/models/quantization/test_nvfp4.py::test_nvfp4[emulation-True-nvidia/Llama-3.1-8B-Instruct-NVFP4]
tests/models/quantization/test_nvfp4.py::test_nvfp4[emulation-False-nvidia/Llama-3.1-8B-Instruct-NVFP4]
tests/models/quantization/test_nvfp4.py::test_nvfp4[flashinfer_cudnn-True-nvidia/Llama-3.1-8B-Instruct-NVFP4]
tests/models/quantization/test_nvfp4.py::test_nvfp4[flashinfer_cudnn-False-nvidia/Llama-3.1-8B-Instruct-NVFP4]
tests/models/quantization/test_nvfp4.py::test_nvfp4[flashinfer_trtllm-True-nvidia/Llama-3.1-8B-Instruct-NVFP4]
tests/models/quantization/test_nvfp4.py::test_nvfp4[flashinfer_trtllm-False-nvidia/Llama-3.1-8B-Instruct-NVFP4]
tests/models/quantization/test_nvfp4.py::test_nvfp4[flashinfer_cutlass-True-nvidia/Llama-3.1-8B-Instruct-NVFP4]
tests/models/quantization/test_nvfp4.py::test_nvfp4[flashinfer_cutlass-False-nvidia/Llama-3.1-8B-Instruct-NVFP4]
  /usr/lib/python3.12/multiprocessing/popen_fork.py:66: DeprecationWarning: This process (pid=7512) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=================================================================== short test summary info ====================================================================
FAILED tests/models/quantization/test_nvfp4.py::test_nvfp4[flashinfer_trtllm-True-nvidia/Llama-3.1-8B-Instruct-NVFP4] - RuntimeError: Engine core initialization failed. See root cause above. Failed core proc(s): {}
=============================================== 1 failed, 7 passed, 3 skipped, 25 warnings in 306.65s (0:05:06) ================================================
sys:1: DeprecationWarning: builtin type swigvarlink has no __module__ attribute
🚨 Error: The command exited with status 1
user command error: exit status 1
```

### 📝 History of failing test

Bisection:
- Last passing build: #66633 (May 18 nightly, commit 23c15acd) — enable_flashinfer_autotune: False for O0/O1/O2
- First failing build: #66759 (May 18 daily, commit cd49a05d) — includes 8c296de6 (PR #42857) which set enable_flashinfer_autotune: True for O1 and O2
- Default optimization level is O2, so autotuning is now enabled by default for all users

### CC List.

@wzhao18 @mgoin 

Tagging since it seems related to https://github.com/vllm-project/vllm/pull/42857.

--- a/vllm/model_executor/warmup/kernel_warmup.py
+++ b/vllm/model_executor/warmup/kernel_warmup.py
@@ -106,6 +106,12 @@ def kernel_warmup(worker: "Worker"):
         )
 
 
+# TODO: remove once FlashInfer upstream fixes the persistent file cache
+# to resolve collisions like `use_8x4_sf_layout=True/False`, which causes
+# invalid tactics to be chosen
+_FLASHINFER_USE_PERSISTENT_CACHE = False
+
+
 def flashinfer_autotune(runner: "GPUModelRunner") -> None:
     """
     Autotune FlashInfer operations.
@@ -121,6 +127,15 @@ def flashinfer_autotune(runner: "GPUModelRunner") -> None:
     """
     import vllm.utils.flashinfer as fi_utils
     from vllm.distributed.parallel_state import get_world_group
+
+    if not _FLASHINFER_USE_PERSISTENT_CACHE:
+        with torch.inference_mode(), fi_utils.autotune():
+            runner._dummy_run(
+                num_tokens=runner.scheduler_config.max_num_batched_tokens,
+                skip_eplb=True,
+                is_profile=True,
+            )
+        return
 
     world = get_world_group()
     is_leader = world.rank_in_group == 0

--- a/vllm/model_executor/warmup/kernel_warmup.py
+++ b/vllm/model_executor/warmup/kernel_warmup.py
@@ -135,6 +135,7 @@ def flashinfer_autotune(runner: "GPUModelRunner") -> None:
                 skip_eplb=True,
                 is_profile=True,
             )
+        get_world_group().barrier()
         return
 
     world = get_world_group()

